### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by kx-chen**
Would like feedback on where to go from here. "It works" but I don't think the code is good enough. :)

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const validGithubUsername = require("./index");
```
Comment (andrewda) [inline test.js]: Capitalize the "H" in GitHub, so: `validGitHubUsername`

File: test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const assert = require('assert');
+
+describe("removes illegal characters", function() {
```
Comment (andrewda) [inline test.js]: I've been using single quotes everywhere else, so just use single quotes in the tests, too 👍

File: test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+   it("removes spaces", function() {
+       let username = validGithubUsername("kx chen");
+       assert.equal(username, "kxchen");
+   });
+
+   it("removes / , . ^ [] {} <> ! @ # $ % & * () _ + = | ? ' characters", function(){
```
Comment (andrewda) [inline test.js]: "removes special characters" would probably be easier to understand

File: test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const validGithubUsername = require("./index");
+const assert = require('assert');
+
+describe("removes illegal characters", function() {
+   it("removes spaces", function() {
```
Comment (andrewda) [inline test.js]: If you could also use ES6 arrow functions, that'd be awesome!

```javascript
it('removes spaces', () => {
   // ....
})
```

**Review by andrewda — CHANGES_REQUESTED** [review]
A few minor changes, but looks awesome overall!

**Comment by kx-chen** [comment]
@andrewda Thanks for the feedback! I believe I've addressed the changes. 

Also, I'm not too sure if the commit messages are okay.... any thoughts?

**Comment by yukiisbored** [comment]
Commit messages are alright :+1:. Personally, I suggest squashing both commits into one since Mocha is a requirement but @andrewda may have a different suggestion.

**Comment by kx-chen** [comment]
@yukiisbored  Thanks! I'll keep that in mind. I've submitted the task for review, and will see what Andrew thinks.

**Review by andrewda — APPROVED** [review]
I typically squash and merge anyways, so this looks all good!

**Pull request merged**